[SE-0376] Clarify backDeployed attribute

xwu opened the pull request:
Based on [forum discussion](https://forums.swift.org/t/backdeployed-is-operating-as-an-override-instead-of-only-for-use-in-unsupported-platforms-versions/68071).

I realized that this document doesn't actually have an introduction: instead, that section is the motivation for the feature; therefore, I transcribed @tshortli's explanation of the feature from the forums with light edits into a new introduction and deleted the subsequent reference to polyfills which was pointed out as misleading to some.

tshortli commented on proposals/0376-function-back-deployment.md:
```diff
@@ -9,6 +9,10 @@
 
 ## Introduction
 
+This proposal introduces a `@backDeployed` attribute to allow ABI-stable libraries to extend the effective availability of their own public APIs to older OSes, where the API isn't actually present in the library that shipped with those OSes. While this resembles a polyfill mechanically, it does not help a module to overlay a polyfill on top of an API from another module.
```
`While this resembles a polyfill mechanically, it does not help a module to overlay a polyfill on top of an API from another module.`

This sentence feels a little out of place to me in the introduction section. Maybe it ought to go in the Future Directions section and be expanded upon a bit? That section could discuss in greater detail the use case of providing polyfills for other module's APIs.

xwu replied:
My takeaway from the discussion is that this clarification is very much needed as a headline sentence: I don't think people will or should need to read down to the future directions to learn that this isn't a polyfill feature.

tshortli replied:
Fair enough! I just think it needs wordsmithing to be part of the Introduction if we want to include it there. The current wording (I think taken verbatim from my forum post?) feels to me like an extremely abrupt introduction of the concept of "polyfill" and needs a bit more elaboration.

xwu commented:
@tshortli Yup, I tried to stick with your words as much as possible, the original goal being to curate writings of which you're the author rather than to insert myself. That said, I totally agree it needed wordsmithing, so I've pushed a revised introduction that I hope better addresses your points (as well as the steering group's). Let me know what you think—this being your proposal text, I want to make sure the text retains your intentions as author.
